src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -91,77 +48,12 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
     void ColorSchemes::OnNavigatedTo(const NavigationEventArgs& e)
     {
-        _State = e.Parameter().as<Editor::ColorSchemesPageNavigationState>();
-        _UpdateColorSchemeList();
-
-        // Initialize our color table view model with 16 dummy colors
-        // so that on a ColorScheme selection change, we can loop through
-        // each ColorTableEntry and just change its color. Performing a
-        // clear and 16 appends doesn't seem to update the color pickers
-        // very accurately.
-        for (uint8_t i = 0; i < TableColorNames.size(); ++i)
-        {
-            const auto& entry{ winrt::make<ColorTableEntry>(i, Windows::UI::Color{ 0, 0, 0, 0 }) };
-            if (i < ColorTableDivider)
-            {
-                _CurrentNonBrightColorTable.Append(entry);
-            }
-            else
-            {
-                _CurrentBrightColorTable.Append(entry);
-            }
-        }
-        _CurrentForegroundColor = winrt::make<ColorTableEntry>(ForegroundColorTag, Windows::UI::Color{ 0, 0, 0, 0 });
-        _CurrentBackgroundColor = winrt::make<ColorTableEntry>(BackgroundColorTag, Windows::UI::Color{ 0, 0, 0, 0 });
-        _CurrentCursorColor = winrt::make<ColorTableEntry>(CursorColorTag, Windows::UI::Color{ 0, 0, 0, 0 });
-        _CurrentSelectionBackgroundColor = winrt::make<ColorTableEntry>(SelectionBackgroundColorTag, Windows::UI::Color{ 0, 0, 0, 0 });
-
-        // Try to look up the scheme that was navigated to. If we find it, immediately select it.
-        const std::wstring lastNameFromNav{ _State.LastSelectedScheme() };
-        const auto it = std::find_if(begin(_ColorSchemeList),
-                                     end(_ColorSchemeList),
-                                     [&lastNameFromNav](const auto& scheme) { return scheme.Name() == lastNameFromNav; });
-
-        if (it != end(_ColorSchemeList))
-        {
-            auto scheme = *it;
-            ColorSchemeComboBox().SelectedItem(scheme);
-        }
```

Comment:
@PankajBhojwani 
This is why [this bug](https://github.com/microsoft/terminal/pull/13179#issuecomment-1139932852) is occurring.

I fixed it using this code snippet:
```cpp
        // Try to look up the scheme that was navigated to. If we find it, immediately select it.
        const auto lastNameFromNav{ _ViewModel.CurrentScheme().Name() };
        const auto it = _ViewModel.AllColorSchemes().First();
        while (it.HasCurrent())
        {
            auto scheme = *it;
            if (scheme.Name() == lastNameFromNav)
            {
                ColorSchemeComboBox().SelectedItem(scheme);
                break;
            }
            it.MoveNext();
        }
```
I tried to use `std::find_if` as before but there seems to be a crash caused by the iterators? Specifically something about comparing the iterators when using `std::find_if` on `begin` and `end`. @lhecker would probably know how to get around it and look spiffy 😉